lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -388,65 +388,127 @@ def compile_mixins(tree, constant)
+                # Similarly, we need this here so that we can handle any
+                # random class method calls on the fake including class
+                # that may be done by the included module during the
+                # `self.included` hook.
               end
             end
             # rubocop:enable Style/MissingRespondToMissing
           end.include(constant)
 
-          all_dynamic_extends = mixins_from_modules.delete(constant)
-          all_dynamic_includes = mixins_from_modules.keys
-          dynamic_extends_from_dynamic_includes = mixins_from_modules.values.flatten
-          dynamic_extends = all_dynamic_extends - dynamic_extends_from_dynamic_includes
-
-          all_dynamic_includes
-            .select { |mod| (name = name_of(mod)) && !name.start_with?("T::") }
-            .map do |mod|
-              add_to_symbol_queue(name_of(mod))
-
-              qname = qualified_name_of(mod)
-              tree << RBI::Include.new(T.must(qname))
-            end.join("\n")
+          [
+            # The value that corresponds to the original included constant
+            # is the list of all dynamically extended modules because of that
+            # constant. We grab that value by deleting the key for the original
+            # constant.
+            T.must(mixins_from_modules.delete(constant)),
+            # Since we deleted the original constant from the list of keys, all
+            # the keys that remain are the ones that are dynamically included modules
+            # during the include of the original constant.
+            mixins_from_modules.keys,
+          ]
+        end
+
+        sig { params(constant: Module, dynamic_extends: T::Array[Module]).returns(T::Array[Module]) }
+        def collect_mixed_in_class_methods(constant, dynamic_extends)
+          if Tapioca::Compilers::Sorbet.supports?(:mixes_in_class_methods_multiple_args)
+            # If we can generate multiple mixes_in_class_methods, then
+            # we want to use all dynamic extends that are not the constant itself
+            return dynamic_extends.select { |mod| mod != constant }
+          end
 
+          # For older Sorbet version, we do an explicit check for an AS::Concern
+          # related ClassMethods module.
           ancestors = singleton_class_of(constant).ancestors
           extends_as_concern = ancestors.any? do |mod|
             qualified_name_of(mod) == "::ActiveSupport::Concern"
           end
           class_methods_module = resolve_constant("#{name_of(constant)}::ClassMethods")
 
           mixed_in_module = if extends_as_concern && Module === class_methods_module
+            # If this module is a concern and the ClassMethods module exists
+            # then, we prefer to generate a mixes_in_class_methods call for
+            # that module only, since we only have a single shot.
             class_methods_module
           else
+            # Otherwise, we use the first dynamic extend module that is not
```

Comment:
In that case, could we create a fictive module that brings all the mixes then mixes this fictive module instead?